src/progressbar/progressbar.js

```diff
@@ -27,6 +27,17 @@ angular.module('ui.bootstrap.progressbar', [])
 
         bar.recalculatePercentage = function() {
             bar.percent = +(100 * bar.value / bar.max).toFixed(2);
+			
+            var totalPercentage = 0;
+            self.bars.forEach(function (bar) {
+                totalPercentage += bar.percent;
+            });
+
+            if (totalPercentage > 100) {
+                bar.percent -= totalPercentage - 100;
+            } else if (totalPercentage < 100) {
```

Comment:
The `else if` should be removed - this is causing the tests to fail. We also should not assume that the bar should fill up the progress section.